src/test/java/com/redhat/devtools/lsp4ij/client/SettingsHelperTest.java

```diff
@@ -0,0 +1,73 @@
+package com.redhat.devtools.lsp4ij.client;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.intellij.testFramework.fixtures.BasePlatformTestCase;
+
+import java.io.StringReader;
+
+public class SettingsHelperTest extends BasePlatformTestCase {
+    // language=json
+    private final String testJson = """
+            {
+                "mylsp": {
+                    "myscalarsetting": "value",
+                    "myobjectsettings": {
+                        "subsettingA": 1,
+                        "subsettingB": 2
+                    }
+                }
+            }
+            """;
+
+    public void testGetSettingsPrimitiveValue() {
+        JsonObject jsonObject = JsonParser.parseReader(new StringReader(testJson)).getAsJsonObject();
```

Comment:
To avoid duplicating code I suggest that you move this load and the assert check in one static methad assertFindSettings(String json, String[] sections, String expected)

Reply:
As far as i understood your suggestion this has now been done. Correct me of i am wrong